### PR TITLE
Add reCAPTCHA protection to DeleteDataForm

### PR DESCRIPTION
**Pull request by holdenk**
## Summary
Add reCAPTCHA v2 protection to the public "delete my data" flow to prevent bot abuse of the data deletion endpoint. Also includes two small incidental fixes needed to unblock CI on this branch.

## Primary change: reCAPTCHA on public delete flow
- Split `DeleteDataForm` into a base (no captcha, used by `AdminDeleteDataView` so authenticated staff don't face friction) and a `PublicDeleteDataForm` subclass used by `RemoveDataView` / `ConfirmDeleteDataView`.
- `PublicDeleteDataForm` defaults to a hidden no-op CharField and swaps in a real `ReCaptchaField` (v2 Checkbox) at instance construction time when Django settings have:
  - `RECAPTCHA_PUBLIC_KEY` set (non-empty)
  - `RECAPTCHA_PRIVATE_KEY` set (non-empty)
  - `RECAPTCHA_TESTING` not enabled
- Gates on `django.conf.settings` (single source of truth) rather than `os.environ` at import time so behavior is overridable via `override_settings` in tests.
- Added sync form-validation tests covering dev/testing mode, reCAPTCHA-enabled mode, and the edge cases where only one key is set or `RECAPTCHA_TESTING` is True.

## Incidental CI fixes (pre-existing failures on main)
These were needed to get CI green on this branch and are not strictly related to reCAPTCHA. Happy to split into a separate PR if preferred:

- **`email_utils.py`**: Added several disposable/temporary email providers (`mailinator.com`, `guerrillamail.com`, `tempmail.com`, `throwaway.email`, `yopmail.com`) to `BLOCKED_EMAIL_DOMAINS`. The existing `tests/async-unit/test_send_fallback_email_filtering.py::test_skips_blocked_domain` test was asserting `mailinator.com` should be blocked but the implementation didn't match. Updated `test_email_utils.py` assertions to agree with the new canonical behavior.
- **`ml/ml_appeal_questions_helper.py`**: `generate_generic_questions` was concatenating `partial_qa_backends() + full_qa_backends()` without deduplication, causing the same model to be called twice and breaking `tests/async-unit/test_generic_model_caching.py::test_generic_question_generation_cache`. Fixed by deduplicating via `dict.fromkeys(...)` while preserving order.

https://claude.ai/code/session_01MAhMt36DbNkUEK2ADYpmbr

### CODE REVIEW

File: fighthealthinsurance/email_utils.py [hunk 1]
```diff
@@ -11,6 +11,12 @@
         "example.org",
         "invalid",
         "test",
+        # Disposable/temporary email providers
+        "mailinator.com",
+        "guerrillamail.com",
+        "tempmail.com",
+        "throwaway.email",
```
[the remark this answers is not in the document]
Comment (holdenk) [reply fighthealthinsurance/email_utils.py:18]: Updated the PR description to call out the expanded disposable-domain blocklist. This change was needed to unblock CI — `tests/async-unit/test_send_fallback_email_filtering.py::test_skips_blocked_domain` was already asserting `mailinator.com` should be blocked, but the implementation didn't match. Happy to split into a separate PR if preferred.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: fighthealthinsurance/ml/ml_appeal_questions_helper.py [hunk 2]
```diff
@@ -31,7 +31,11 @@ async def generate_generic_questions(
         Returns:
             A list of (question, answer) tuples.
         """
-        models_to_try = ml_router.partial_qa_backends() + ml_router.full_qa_backends()
+        models_to_try = list(
+            dict.fromkeys(
+                ml_router.partial_qa_backends() + ml_router.full_qa_backends()
+            )
+        )
```
[the remark this answers is not in the document]
Comment (holdenk) [reply fighthealthinsurance/ml/ml_appeal_questions_helper.py:38]: Updated the PR description to call out the dedup change. This was needed to unblock CI — `test_generic_question_generation_cache` mocks both `partial_qa_backends` and `full_qa_backends` to return the same model instance, so the pre-existing concatenation caused `get_appeal_questions` to be called twice. Happy to split into a separate PR if preferred.

---
_Generated by [Claude Code](https://claude.ai/code)_
